### PR TITLE
Ball

### PR DESCRIPTION
**Pull request by BallballTH**
Add:
- Sound can now be changed when switched Scene (if theres any)
- More sound when died
- More colors on Texts
- function Print ---- depends on terminal's width
- a little delay before print texts to prevent some "shifted text" (sometimes)
- Playagain? when in Ending scene

Fix:
- ImportFile function (statchange)
- ResetSaveFile function

Adjust:
- Print scene,hp sa bar
- dialogues

### CODE REVIEW

**Comment by BallballTH** [comment]
Ex.

Screenshot 2024-03-04 031309

File: Dialogue/Options.txt [hunk 1]
```diff
@@ -172,4 +172,14 @@ prologue|พิมพ์เลข 1 เพื่อดำเนินเรื
...
+ending_you_die|[g]Play again[\]|prologue|
+ending_you_die|[r]Leave[\]|Leave|
+ending_execution|[g]Play again[\]|prologue|
+ending_execution|[r]Leave[\]|Leave|
+ending1|[g]Play again[\]|prologue|
+ending1|[r]Leave[\]|Leave|
+ending2|[g]Play again[\]|prologue|
+ending2|[r]Leave[\]|Leave|
+ending3|[g]Play again[\]|prologue|
+ending3|[r]Leave[\]|Leave|
```
Comment (OrneryCash) [inline Dialogue/Options.txt]: \ ?

File: dialogueMaker/dialogueMaker.cpp [hunk 2]
```diff
@@ -224,8 +250,17 @@ std::string Game::parseText(std::string text) {
   std::map<std::string, std::string> decorations = {
     {"b", "\033[1m"}, // Bold
     {"ul", "\033[4m"}, // Underline
-    {"r", "\033[31m"}, // Red
-    {"g", "\033[32m"}, // Green
+    {"br", "\033[38;2;57;21;0m"}, // brown
+    {"bl", "\033[38;2;17;85;204m"}, // blue
+    {"bla", "\033[38;2;.0;0;0m"}, // black
```
Comment (OrneryCash) [inline dialogueMaker/dialogueMaker.cpp]:
```suggestion
    {"bla", "\033[38;2;0;0;0m"}, // black
```

**Review by OrneryCash — CHANGES_REQUESTED** [review]

File: Dialogue/Options.txt (hunk 1 above)
Comment (BallballTH) [reply Dialogue/Options.txt]: mb

**Review by OrneryCash — APPROVED** [review]
cool

**Pull request merged**